lms/static/scripts/frontend_apps/components/test/LMSFilePicker-test.js

```diff
@@ -62,18 +89,103 @@ describe('LMSFilePicker', () => {
 
   it('fetches files when the dialog first appears', async () => {
     const wrapper = renderFilePicker();
+
     assert.calledWith(fakeApiCall, {
       authToken: 'auth-token',
       path: fakeListFilesApi.path,
     });
-    const returnedFiles = await fakeApiCall.returnValues[0];
-    // Component currently filters out Folders
-    const expectedFiles = returnedFiles.filter(file => file.type !== 'Folder');
+
+    const expectedFiles = await fakeApiCall.returnValues[0];
     wrapper.update();
+
     const fileList = wrapper.find('FileList');
     assert.deepEqual(fileList.prop('files'), expectedFiles);
   });
 
+  it('shows breadcrumbs if `withBreadcrumbs` enabled', async () => {
+    const wrapper = renderFilePicker({ withBreadcrumbs: true });
+
+    // Breadcrumbs will only render once the fetch has completed
+    await fakeApiCall;
+    wrapper.update();
+
+    assert.include(
+      wrapper.find('Breadcrumbs').props().items[0],
+      {
+        display_name: 'Files',
+        id: '__root__',
+      },
+      'The only initial breadcrumb is the top-level "Files" crumb'
+    );
+
+    assert.equal(
+      wrapper.find('Breadcrumbs').props().renderItem(fakeFolders[0]),
+      'Subfolder',
+      'The `renderItem` callback passed to Breadcrumbs renders a File `display_name`'
+    );
+  });
+
+  it('fetches files in indicated sub-folder if folder path is changed', async () => {
+    const wrapper = renderFilePicker({ withBreadcrumbs: true });
+
+    // First, automatic fetch
+    await fakeApiCall;
+    wrapper.update();
+
+    fakeApiCall.reset();
+    fakeApiCall.resolves(fakeFiles);
+
+    // Simulate changing the folder path, as if a user clicked on a "crumb"
+    await act(() =>
+      wrapper.find('Breadcrumbs').props().onSelectItem(fakeFolders[0])
+    );
+
+    await fakeApiCall;
+
+    wrapper.update();
```

Comment:
The `wrapper.update()` call is not strictly needed here as you're not checking the updated output.